designdocs/capacity_staking_rewards_implementation.md

```diff
@@ -0,0 +1,380 @@
+
+# Problem Statement
+This document outlines how to implement the Staking for Rewards feature described in [Capacity Staking Rewards Economic Model (TBD)](TBD), without, at this time, regard to what the economic model actually is.
+
+## Glossary
+1. **FRQCY**: the native token of the Frequency blockchain
+1. **Capacity**: the non-transferrable utility token which can be used only to pay for certain Frequency transactions.
+1. **Account**: a Frequency System Account controlled by a private key and addressed by a public key, having at least a minimum balance (currently 0.01 FRQCY).
+1. **Stake** (verb): to lock some amount of a token against transfer for a period of time in exchange for some reward.
+1. **RewardEra**: the time period (TBD in blocks or Capacity Epochs) that Staking Rewards are based upon. RewardEra is to distinguish it easily from Substrate's staking pallet Era, or the index of said time period.
+1. **Staking Reward**: a per-RewardEra share of a staking reward pool of FRQCY tokens for a given staking account.
+1. **Reward Pool**:  a fixed amount of FRQCY that can be minted for rewards each RewardEra and distributed to stakers.
+1. **StakingRewardsProvider**: a trait that encapsulates the economic model for staking rewards, providing functionality for calculating the reward pool and staking rewards.
+
+## Staking Token Rewards
+
+### StakingAccountDetails updates
+New fields are added. The field **`last_rewarded_at`** is to keep track of the last time rewards were claimed for this Staking Account.
+MaximumCapacity staking accounts MUST always have the value `None` for `last_rewarded_at`. 
+Finally, `stake_change_unlocking`, a BoundedVec is added which tracks the chunks of when a staking account has changed 
+targets for some amount of funds.  This is to prevent retarget spamming.
+```rust
+pub struct StakingAccountDetails {
+    pub active: BalanceOf<T>,
+    pub total: BalanceOf<T>,
+    pub unlocking: BoundedVec<UnlockChunk<BalanceOf<T>, T::EpochNumber>, T::MaxUnlockingChunks>,
+    /// The number of the last StakingEra that this account's rewards were claimed.
+    pub last_rewards_claimed_at: Option<T::RewardEra>, // NEW  None means never rewarded, Some(RewardEra) means last rewarded RewardEra.
+    /// staking amounts that have been retargeted are prevented from being retargeted again for the
+    /// configured Thawing Period number of blocks.
+    pub stake_change_unlocking: BoundedVec<UnlockChunk<BalanceOf<T>, T::RewardEra>, T::MaxUnlockingChunks>, // NEW
+}
+```
+
+### StakingTargetDetails updates, StakingHistory
+A new field, `staking_type` is added to indicate the type of staking the Account holder is doing in relation to this target.
+Staking type may be `MaximumCapacity` or `ProviderBoost`. `MaximumCapacity` is the default value for `staking_type` and maps to 0.
```

Comment:
This definition seems redundant with what is later described under `staking_type`.